# Веб-скрапинг. ДЗ в рамках курса Scypro "Аналитик данных"


# ✏️ 🗒️ 📝

Сегодня тебе нужно выполнить задачу от руководительницы отдела контент-маркетинга новой онлайн-школы английского языка. Она хочет составить ТЗ копирайтеру на написание постов, но для этого ей нужно знать, какие типы постов популярны. За завтраком вы с ней обсуждали твое последнее занятие с курса Python, где мы парсили страницы сообществ ВК, и у вас появилась идея – проанализировать самые успешные посты известной онлайн-школы Skyeng  в [сообществе Skyeng в ВК](https://vk.com/skyeng).

Теперь перед тобой стоит задача – выгрузить маркетологам для анализа 50 самых популярных постов за все время существования сообщества. Метрику популярности будем считать как сумму лайков, репостов и комментариев к каждому посту. Если останется время, для дополнительного анализа выгрузи все комментарии к этим постам в отдельный файл.

## Задание 1

Получи собственный `access_token` в ВК. 

In [1]:
# читаем токен из файла my_own_token.py, куда мы сохранили access_token
from my_own_token2 import my_own_token2 as token

## Задание 2

__Сформируй запрос, который получает 100 записей со стены сообщества__


In [2]:
import requests 
import pandas as pd

In [3]:
url = 'https://api.vk.com/method/wall.get'

params = {'access_token': token,
          'v':'5.89',
          'domain': 'skyeng',
          'count':100, # берем 100 записей
          'filter': 'all'  # Берём все посты
            
        }
req = requests.get(url, params)
res = req.json()

print('Кол-во записей:', len(pd.DataFrame(res['response'])))

Кол-во записей: 100


In [4]:
res

{'response': {'count': 7921,
  'items': [{'id': 86063,
    'from_id': -51890028,
    'owner_id': -51890028,
    'date': 1643305415,
    'marked_as_ads': 0,
    'post_type': 'post',
    'text': 'Если вы интроверт, то большая часть советов по изучению языка звучит как издевка. Общайтесь с новыми людьми, сходите в разговорный клуб — но что, если вы не хотите общаться с людьми? Собрали несколько советов для интровертов, которые мечтают подтянуть навыки разговорного английского.\n\nКомфортное обучение для всех темпераментов со скидкой до 45% на киберраспродаже по ссылке в шапке профиля до 31 января 😉\nhttps://go.skyeng.ru/cyb',
    'attachments': [{'type': 'photo',
      'photo': {'album_id': -7,
       'date': 1643305415,
       'id': 457279226,
       'owner_id': -51890028,
       'access_key': '9b3fcc6131890871fe',
       'sizes': [{'height': 130,
         'url': 'https://sun9-36.userapi.com/impg/-8rExLQxwQ1-cUA58AGzTTbcSp_ldhH3GRRcZA/2cBUWlGKVS0.jpg?size=104x130&quality=96&sign=def571ea

## Задание 3

Выгрузи в переменную `posts` в формате датафрейма __все записи__ сообщества Skyeng: https://vk.com/skyeng, 


In [5]:
#соберем все записи сообщества 
import time # для sleep

offset = 0
max_count = 100
posts = pd.DataFrame()

while True:      # Повторять цикл пока его не остановят по условию ниже
    
    url = 'https://api.vk.com/method/wall.get'
    params = {
        'access_token': token,
        'v':'5.89',
        'domain': 'skyeng',
        'count': max_count,
        'offset': offset
    }
    
    req = requests.get(url=url, 
                       params=params)
    res = req.json()
    
    if len(res['response']['items']) != 0:      # Проверяем вернулись ли нам хоть какие-нибудь записи
        posts = pd.concat([posts, pd.DataFrame(res['response']['items'])]) 
    else:
        break                                   # Если нам ничего не вернулось, то прервать цикл
    print('Уже скачано', offset, 'записей')
    offset = offset + max_count                 # Смещаем offset на число возвращенных записей
    time.sleep(0.5)                             # Ограничиваем частоту запросов до одного запроса в полсекунды

Уже скачано 0 записей
Уже скачано 100 записей
Уже скачано 200 записей
Уже скачано 300 записей
Уже скачано 400 записей
Уже скачано 500 записей
Уже скачано 600 записей
Уже скачано 700 записей
Уже скачано 800 записей
Уже скачано 900 записей
Уже скачано 1000 записей
Уже скачано 1100 записей
Уже скачано 1200 записей
Уже скачано 1300 записей
Уже скачано 1400 записей
Уже скачано 1500 записей
Уже скачано 1600 записей
Уже скачано 1700 записей
Уже скачано 1800 записей
Уже скачано 1900 записей
Уже скачано 2000 записей
Уже скачано 2100 записей
Уже скачано 2200 записей
Уже скачано 2300 записей
Уже скачано 2400 записей
Уже скачано 2500 записей
Уже скачано 2600 записей
Уже скачано 2700 записей
Уже скачано 2800 записей
Уже скачано 2900 записей
Уже скачано 3000 записей
Уже скачано 3100 записей
Уже скачано 3200 записей
Уже скачано 3300 записей
Уже скачано 3400 записей
Уже скачано 3500 записей
Уже скачано 3600 записей
Уже скачано 3700 записей
Уже скачано 3800 записей
Уже скачано 3900 записей
Уже скачано 

In [6]:
print('Кол-во записей:', len(posts))

Кол-во записей: 7921


In [7]:
posts.head()

id   from_id  owner_id        date  marked_as_ads post_type  \
0  86063 -51890028 -51890028  1643305415              0      post   
1  86062 -51890028 -51890028  1643220435              0      post   
2  86055 -51890028 -51890028  1643144206              0      post   
3  86054 -51890028 -51890028  1643029717              0      post   
4  86051 -51890028 -51890028  1642852419              0      post   

                                                text  \
0  Если вы интроверт, то большая часть советов по...   
1  Неважно, почему вы рассматриваете переезд в за...   
2  Дагеротипист, фонарщик, телефонистка. Вполне с...   
3  🤖 Киберпонедельник со скидками до 45% уже нача...   
4  Учитесь в Skyeng, чтобы понимать тонкости англ...   

                                         attachments  \
0  [{'type': 'photo', 'photo': {'album_id': -7, '...   
1  [{'type': 'photo', 'photo': {'album_id': -7, '...   
2  [{'type': 'photo', 'photo': {'album_id': -7, '...   
3  [{'type': 'photo', 'photo': {'album_id': -7, '...   
4  [{'type': 'photo', 'photo': {'album_id': -7, '...   

                             post_source  \
0                         {'type': 'vk'}   
1                         {'type': 'vk'}   
2  {'platform': 'iphone', 'type': 'api'}   
3  {'platform': 'iphone', 'type': 'api'}   
4                         {'type': 'vk'}   

                                            comments  \
0  {'can_post': 1, 'count': 4, 'groups_can_post':...   
1  {'can_post': 1, 'count': 0, 'groups_can_post':...   
2  {'can_post': 1, 'count': 8, 'groups_can_post':...   
3  {'can_post': 1, 'count': 0, 'groups_can_post':...   
4  {'can_post': 1, 'count': 2, 'groups_can_post':...   

                                               likes  \
0  {'can_like': 1, 'count': 31, 'user_likes': 0, ...   
1  {'can_like': 1, 'count': 20, 'user_likes': 0, ...   
2  {'can_like': 1, 'count': 13, 'user_likes': 0, ...   
3  {'can_like': 1, 'count': 6, 'user_likes': 0, '...   
4  {'can_like': 1, 'count': 34, 'user_likes': 0, ...   

                             reposts            views  \
0  {'count': 22, 'user_reposted': 0}  {'count': 3625}   
1  {'count': 17, 'user_reposted': 0}  {'count': 4415}   
2  {'count': 10, 'user_reposted': 0}  {'count': 5872}   
3   {'count': 0, 'user_reposted': 0}  {'count': 4608}   
4  {'count': 11, 'user_reposted': 0}  {'count': 7266}   

                          hash copy_history  signer_id  geo  
0  1_wJEF-seFaz3aQ45YNEHj9JXk4          NaN        NaN  NaN  
1  prgcJQ03MIIvzL0WcCgwHJPYLsU          NaN        NaN  NaN  
2  iJSaCLnkcv3jODX_LTpBaMZt7cM          NaN        NaN  NaN  
3  RFBlsuvm4KDCnQbVYmgTsDIunVU          NaN        NaN  NaN  
4  CpfE4EYrJNkQJ1dWeNfmGc_cf1I          NaN        NaN  NaN

## Задание 4

Выгрузи 50 самых популярных постов. Метрикой популярности мы считаем сумму лайков, репостов и комментариев.


In [8]:
# вытащим из словарей столбцов comments, likes, reposts только нужные значения
posts_copy = posts.copy() #сохраним сырые данные
def get_value(name_column):
    posts_copy[name_column + '_count'] = posts_copy[name_column]
    for i in range(0, len(posts_copy)):
        posts_copy[name_column + '_count'].iloc[i] = posts_copy[name_column].iloc[i]['count']


    


In [9]:
column_name_list = ['comments', 'likes', 'reposts']
for name in column_name_list:
    get_value(name)
    


C:\ProgramData\Anaconda3\lib\site-packages\pandas\core\indexing.py:1637: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_block(indexer, value, name)


In [10]:
posts_copy.head()

id   from_id  owner_id        date  marked_as_ads post_type  \
0  86063 -51890028 -51890028  1643305415              0      post   
1  86062 -51890028 -51890028  1643220435              0      post   
2  86055 -51890028 -51890028  1643144206              0      post   
3  86054 -51890028 -51890028  1643029717              0      post   
4  86051 -51890028 -51890028  1642852419              0      post   

                                                text  \
0  Если вы интроверт, то большая часть советов по...   
1  Неважно, почему вы рассматриваете переезд в за...   
2  Дагеротипист, фонарщик, телефонистка. Вполне с...   
3  🤖 Киберпонедельник со скидками до 45% уже нача...   
4  Учитесь в Skyeng, чтобы понимать тонкости англ...   

                                         attachments  \
0  [{'type': 'photo', 'photo': {'album_id': -7, '...   
1  [{'type': 'photo', 'photo': {'album_id': -7, '...   
2  [{'type': 'photo', 'photo': {'album_id': -7, '...   
3  [{'type': 'photo', 'photo': {'album_id': -7, '...   
4  [{'type': 'photo', 'photo': {'album_id': -7, '...   

                             post_source  \
0                         {'type': 'vk'}   
1                         {'type': 'vk'}   
2  {'platform': 'iphone', 'type': 'api'}   
3  {'platform': 'iphone', 'type': 'api'}   
4                         {'type': 'vk'}   

                                            comments  \
0  {'can_post': 1, 'count': 4, 'groups_can_post':...   
1  {'can_post': 1, 'count': 0, 'groups_can_post':...   
2  {'can_post': 1, 'count': 8, 'groups_can_post':...   
3  {'can_post': 1, 'count': 0, 'groups_can_post':...   
4  {'can_post': 1, 'count': 2, 'groups_can_post':...   

                                               likes  \
0  {'can_like': 1, 'count': 31, 'user_likes': 0, ...   
1  {'can_like': 1, 'count': 20, 'user_likes': 0, ...   
2  {'can_like': 1, 'count': 13, 'user_likes': 0, ...   
3  {'can_like': 1, 'count': 6, 'user_likes': 0, '...   
4  {'can_like': 1, 'count': 34, 'user_likes': 0, ...   

                             reposts            views  \
0  {'count': 22, 'user_reposted': 0}  {'count': 3625}   
1  {'count': 17, 'user_reposted': 0}  {'count': 4415}   
2  {'count': 10, 'user_reposted': 0}  {'count': 5872}   
3   {'count': 0, 'user_reposted': 0}  {'count': 4608}   
4  {'count': 11, 'user_reposted': 0}  {'count': 7266}   

                          hash copy_history  signer_id  geo comments_count  \
0  1_wJEF-seFaz3aQ45YNEHj9JXk4          NaN        NaN  NaN              4   
1  prgcJQ03MIIvzL0WcCgwHJPYLsU          NaN        NaN  NaN              0   
2  iJSaCLnkcv3jODX_LTpBaMZt7cM          NaN        NaN  NaN              8   
3  RFBlsuvm4KDCnQbVYmgTsDIunVU          NaN        NaN  NaN              0   
4  CpfE4EYrJNkQJ1dWeNfmGc_cf1I          NaN        NaN  NaN              2   

  likes_count reposts_count  
0          31            22  
1          20            17  
2          13            10  
3           6             0  
4          34            11

In [11]:
# заполняем даными через lambda функцию
posts['comments_count'] = posts['comments'].apply(lambda x: x.get('count'))
posts['likes_count'] = posts['likes'].apply(lambda x: x.get('count'))
posts['reposts_count'] = posts['reposts'].apply(lambda x: x.get('count'))
posts.head()

id   from_id  owner_id        date  marked_as_ads post_type  \
0  86063 -51890028 -51890028  1643305415              0      post   
1  86062 -51890028 -51890028  1643220435              0      post   
2  86055 -51890028 -51890028  1643144206              0      post   
3  86054 -51890028 -51890028  1643029717              0      post   
4  86051 -51890028 -51890028  1642852419              0      post   

                                                text  \
0  Если вы интроверт, то большая часть советов по...   
1  Неважно, почему вы рассматриваете переезд в за...   
2  Дагеротипист, фонарщик, телефонистка. Вполне с...   
3  🤖 Киберпонедельник со скидками до 45% уже нача...   
4  Учитесь в Skyeng, чтобы понимать тонкости англ...   

                                         attachments  \
0  [{'type': 'photo', 'photo': {'album_id': -7, '...   
1  [{'type': 'photo', 'photo': {'album_id': -7, '...   
2  [{'type': 'photo', 'photo': {'album_id': -7, '...   
3  [{'type': 'photo', 'photo': {'album_id': -7, '...   
4  [{'type': 'photo', 'photo': {'album_id': -7, '...   

                             post_source  \
0                         {'type': 'vk'}   
1                         {'type': 'vk'}   
2  {'platform': 'iphone', 'type': 'api'}   
3  {'platform': 'iphone', 'type': 'api'}   
4                         {'type': 'vk'}   

                                            comments  \
0  {'can_post': 1, 'count': 4, 'groups_can_post':...   
1  {'can_post': 1, 'count': 0, 'groups_can_post':...   
2  {'can_post': 1, 'count': 8, 'groups_can_post':...   
3  {'can_post': 1, 'count': 0, 'groups_can_post':...   
4  {'can_post': 1, 'count': 2, 'groups_can_post':...   

                                               likes  \
0  {'can_like': 1, 'count': 31, 'user_likes': 0, ...   
1  {'can_like': 1, 'count': 20, 'user_likes': 0, ...   
2  {'can_like': 1, 'count': 13, 'user_likes': 0, ...   
3  {'can_like': 1, 'count': 6, 'user_likes': 0, '...   
4  {'can_like': 1, 'count': 34, 'user_likes': 0, ...   

                             reposts            views  \
0  {'count': 22, 'user_reposted': 0}  {'count': 3625}   
1  {'count': 17, 'user_reposted': 0}  {'count': 4415}   
2  {'count': 10, 'user_reposted': 0}  {'count': 5872}   
3   {'count': 0, 'user_reposted': 0}  {'count': 4608}   
4  {'count': 11, 'user_reposted': 0}  {'count': 7266}   

                          hash copy_history  signer_id  geo  comments_count  \
0  1_wJEF-seFaz3aQ45YNEHj9JXk4          NaN        NaN  NaN               4   
1  prgcJQ03MIIvzL0WcCgwHJPYLsU          NaN        NaN  NaN               0   
2  iJSaCLnkcv3jODX_LTpBaMZt7cM          NaN        NaN  NaN               8   
3  RFBlsuvm4KDCnQbVYmgTsDIunVU          NaN        NaN  NaN               0   
4  CpfE4EYrJNkQJ1dWeNfmGc_cf1I          NaN        NaN  NaN               2   

   likes_count  reposts_count  
0           31             22  
1           20             17  
2           13             10  
3            6              0  
4           34             11

In [12]:
# заполняем данными столбец с суммой по всем активностям
posts['activity'] = posts[['comments_count','likes_count', 'reposts_count']].sum(axis=1)

In [13]:
#преобразуем дату поста к привычному формату
from datetime import datetime
posts['date'] = posts['date'].apply(lambda x: datetime.utcfromtimestamp(x).strftime('%Y-%m-%d %H:%M:%S'))

In [14]:
#выбираем 50 самыx популярныx 
posts_top50 = posts.sort_values(by = 'activity', ascending = False).reset_index().head(50)
posts_top50.head()

index     id   from_id  owner_id                 date  marked_as_ads  \
0     20  63081 -51890028 -51890028  2020-10-13 12:00:09              0   
1     67  25259 -51890028 -51890028  2019-04-09 08:57:24              0   
2     14    977 -51890028 -51890028  2014-11-25 15:41:49              0   
3     55  75933 -51890028 -51890028  2020-11-09 12:00:04              0   
4     46  42960 -51890028 -51890028  2019-09-14 06:57:29              0   

  post_type                                               text  \
0      post  КОНКУРС! Выиграйте уроки английского и бьюти-б...   
1      post  Через 15 дней вы накачаете пресс и выучите анг...   
2      post  100 полезных английских фраз\nЭти фразы выруча...   
3      post  КОНКУРС! Выиграйте уроки английского и боксы с...   
4      post  😑Предлоги в английском — больная тема для всех...   

                                         attachments     post_source  ...  \
0  [{'type': 'photo', 'photo': {'album_id': -7, '...  {'type': 'vk'}  ...   
1  [{'type': 'photo', 'photo': {'album_id': -7, '...  {'type': 'vk'}  ...   
2  [{'type': 'photo', 'photo': {'album_id': -7, '...  {'type': 'vk'}  ...   
3  [{'type': 'photo', 'photo': {'album_id': -7, '...  {'type': 'vk'}  ...   
4  [{'type': 'photo', 'photo': {'album_id': -7, '...  {'type': 'vk'}  ...   

                               reposts              views  \
0     {'count': 2, 'user_reposted': 0}   {'count': 15319}   
1  {'count': 1455, 'user_reposted': 0}  {'count': 504231}   
2   {'count': 975, 'user_reposted': 0}                NaN   
3     {'count': 6, 'user_reposted': 0}   {'count': 13201}   
4   {'count': 434, 'user_reposted': 0}  {'count': 177071}   

                          hash copy_history signer_id  geo  comments_count  \
0  bUWaxZGiwvE0_WKA2eUKs80s_tY          NaN       NaN  NaN           10962   
1  tB7dN68_EueDZCVspl8-Rr-ocZY          NaN       NaN  NaN             239   
2  aPJEuWG8jZXtWPuyK0gEacA9IcM          NaN       NaN  NaN               0   
3  OA82EwnWKV35hrgip7hoIaqhJlM          NaN       NaN  NaN            2891   
4  -1u7cY0dz89Ay60_dutEVptnghY          NaN       NaN  NaN              50   

  likes_count  reposts_count  activity  
0         117              2     11081  
1        6994           1455      8688  
2        4012            975      4987  
3         133              6      3030  
4        1917            434      2401  

[5 rows x 22 columns]

## Задание 5

__Почисти данные, сохрани результат в файл для отправки копирайтерам__


In [15]:
#оставляем только нужные столбцы
posts_top50_ = posts_top50[['date', 'text', 'comments_count', 'likes_count', 'reposts_count', 'activity']]
posts_top50_.head()

date                                               text  \
0  2020-10-13 12:00:09  КОНКУРС! Выиграйте уроки английского и бьюти-б...   
1  2019-04-09 08:57:24  Через 15 дней вы накачаете пресс и выучите анг...   
2  2014-11-25 15:41:49  100 полезных английских фраз\nЭти фразы выруча...   
3  2020-11-09 12:00:04  КОНКУРС! Выиграйте уроки английского и боксы с...   
4  2019-09-14 06:57:29  😑Предлоги в английском — больная тема для всех...   

   comments_count  likes_count  reposts_count  activity  
0           10962          117              2     11081  
1             239         6994           1455      8688  
2               0         4012            975      4987  
3            2891          133              6      3030  
4              50         1917            434      2401

In [16]:
posts_top50_.to_csv('posts_top50_.csv', index=False)

## EXTRA

__К 50 самым популярным постам выгрузи все комментарии, чтобы оценить, как пользователи восприняли посты__


In [17]:
posts_id_top50 = posts_top50[['id', 'date', 'text', 'comments_count', 'likes_count', 'reposts_count', 'activity']]

In [18]:
posts_id_top50.head()

id                 date  \
0  63081  2020-10-13 12:00:09   
1  25259  2019-04-09 08:57:24   
2    977  2014-11-25 15:41:49   
3  75933  2020-11-09 12:00:04   
4  42960  2019-09-14 06:57:29   

                                                text  comments_count  \
0  КОНКУРС! Выиграйте уроки английского и бьюти-б...           10962   
1  Через 15 дней вы накачаете пресс и выучите анг...             239   
2  100 полезных английских фраз\nЭти фразы выруча...               0   
3  КОНКУРС! Выиграйте уроки английского и боксы с...            2891   
4  😑Предлоги в английском — больная тема для всех...              50   

   likes_count  reposts_count  activity  
0          117              2     11081  
1         6994           1455      8688  
2         4012            975      4987  
3          133              6      3030  
4         1917            434      2401

In [141]:
#общее количество комментариев
posts_id_top50['comments_count'].sum()

18097

In [38]:
#список id
list_posts_id = posts_id_top50['id'].tolist()


In [39]:
#сформируем тестовый датафрейм для проверки корректности работы цикла
posts_id_top2 = posts_id_top50.iloc[0:11]
list_posts_id2 = posts_id_top2['id'].tolist() #маленький список, который пригодится для проверки цикла
posts_id_top2['comments_count'].sum()

16087

In [42]:
#соберем для каждого поста все комментарии и лайки к ним

owner_id = str(posts['owner_id'].iloc[0])
offset = 0
max_count_comments = 100
comments = pd.DataFrame(columns = {'post_id'})

i = 0

while True:
    periods=0
    url = 'https://api.vk.com/method/wall.getComments'

    params = {'access_token': token,
                'v':'5.89',
                'owner_id': owner_id,
                'post_id': list_posts_id[i],
                'need_likes': 1, 
                'offset': offset,
                'count': max_count,
                'sort': 'desc',
                'preview_length': 0 #не обрезаем комментарии
            }
    req = requests.get(url, params)
    res = req.json()
     
    

    if len(res['response']['items']) != 0: 
        df = pd.DataFrame(res['response']['items'])
        comments = pd.concat([comments, df])

        comments['post_id'] = comments['post_id'].shift(periods=periods) #добавим post_id со смещением 
        comments['post_id'] = comments['post_id'].fillna(str(list_posts_id[i]))
        offset = offset + max_count_comments                 # Смещаем offset на число возвращенных записей
        periods = periods + len(df) #смещение для poat_id
        time.sleep(0.5)
    else:
        if i < len(list_posts_id)-1:
            offset = 0
            periods = 0
            i +=1

        else:
            break
    
    


In [49]:
comments.head()

post_id            id     from_id                 date  \
0   63081  2.000000e+09       100.0  2022-01-28 12:16:44   
1   63081  7.477200e+04  61407702.0  2020-10-20 19:47:26   
2   63081  7.477100e+04  61407702.0  2020-10-20 19:47:20   
3   63081  7.477000e+04  61407702.0  2020-10-20 19:47:14   
4   63081  7.475400e+04 -51890028.0  2020-10-20 16:29:59   

                                                text  \
0  Ваше приложение не поддерживает новую структур...   
1                                                  +   
2                                                  +   
3                                                  +   
4  Всем спасибо за участие! ❤ Итоги конкурса — по...   

                                          likes attachments  reply_to_user  \
0  {'can_like': 1, 'count': 0, 'user_likes': 0}         NaN            NaN   
1  {'can_like': 1, 'count': 0, 'user_likes': 0}         NaN            NaN   
2  {'can_like': 1, 'count': 0, 'user_likes': 0}         NaN            NaN   
3  {'can_like': 1, 'count': 0, 'user_likes': 0}         NaN            NaN   
4  {'can_like': 1, 'count': 0, 'user_likes': 0}         NaN            NaN   

   reply_to_comment  likes_count  
0               NaN            0  
1               NaN            0  
2               NaN            0  
3               NaN            0  
4               NaN            0

In [44]:
posts_id_top50['comments_count'].sum() - len(comments) 

633

Количество выгруженных комментариев меньше на 633, поскольку среди комментариев к посту с id 31067 есть ветки комментариев не к посту, а к самим комментариям, видимо они и не подгружаются(об этом указано и в первой строке датафрейма(приложение не поддерживает текущую структуру комментариев))

In [41]:
#на маленькой выборке видно, что по комментариям выпадает пост 31067
posts_id_top2

id                 date  \
0   63081  2020-10-13 12:00:09   
1   25259  2019-04-09 08:57:24   
2     977  2014-11-25 15:41:49   
3   75933  2020-11-09 12:00:04   
4   42960  2019-09-14 06:57:29   
5   48883  2019-11-22 09:25:57   
6   31067  2019-06-23 14:55:35   
7   55740  2020-04-18 11:00:00   
8   51793  2020-02-07 06:43:07   
9   57606  2020-05-13 12:31:02   
10  60584  2020-07-30 10:26:49   

                                                 text  comments_count  \
0   КОНКУРС! Выиграйте уроки английского и бьюти-б...           10962   
1   Через 15 дней вы накачаете пресс и выучите анг...             239   
2   100 полезных английских фраз\nЭти фразы выруча...               0   
3   КОНКУРС! Выиграйте уроки английского и боксы с...            2891   
4   😑Предлоги в английском — больная тема для всех...              50   
5   Кот Виктор показывает, как стал причиной сканд...              54   
6   Играем в слова. У кого словарный запас больше?...            1634   
7   Без фразовых глаголов можно понимать только са...              36   
8   Английский снова сводит нас с ума, 😉и ему вооб...             111   
9   Готовы ли вы к жизни в англоговорящем городе? ...              84   
10            Подписывайтесь на [club51890028|Skyeng]              26   

    likes_count  reposts_count  activity  
0           117              2     11081  
1          6994           1455      8688  
2          4012            975      4987  
3           133              6      3030  
4          1917            434      2401  
5          1620            210      1884  
6           135              3      1772  
7          1397            275      1708  
8          1383            198      1692  
9          1479            124      1687  
10         1456            159      1641

In [32]:
#выгрузка комментариев при маленькой выборке
comments.groupby('post_id').count()

id  from_id   date   text  likes  attachments  reply_to_user  \
post_id                                                                    
25259      239      239    239    239    239           17            116   
31067     1000     1000   1000   1000   1000            2              0   
42960       50       50     50     50     50            1             18   
48883       54       54     54     54     54            7             31   
51793      111      111    111    111    111            1             13   
55740       36       36     36     36     36            2             23   
57606       84       84     84     84     84            6             54   
60584       26       26     26     26     26            2             14   
63081    10963    10963  10963  10963  10963           17              0   
75933     2891     2891   2891   2891   2891           16              0   

         reply_to_comment  
post_id                    
25259                 116  
31067                   0  
42960                  18  
48883                  31  
51793                  13  
55740                  23  
57606                  54  
60584                  14  
63081                   0  
75933                   0

In [45]:
#преобразуем дату к привычному формату
comments['date'] = comments['date'].apply(lambda x: datetime.utcfromtimestamp(x).strftime('%Y-%m-%d %H:%M:%S'))

In [46]:
#в отдельную колонку запишем количество лайков
comments['likes_count'] = comments['likes']
comments['likes_count'] = comments['likes_count'].apply(lambda x: x.get('count'))

In [48]:
#соберем датафрейм
comments_info = comments[['post_id', 'date', 'text','likes_count']].reset_index()
comments_info = comments_info.iloc[1:] #убираем строку, с тектом, что приложение не поддерживает текущую структуру комментариев
comments_info.head()

index post_id                 date  \
1      1   63081  2020-10-20 19:47:26   
2      2   63081  2020-10-20 19:47:20   
3      3   63081  2020-10-20 19:47:14   
4      4   63081  2020-10-20 16:29:59   
5      5   63081  2020-10-20 16:27:13   

                                                text  likes_count  
1                                                  +            0  
2                                                  +            0  
3                                                  +            0  
4  Всем спасибо за участие! ❤ Итоги конкурса — по...            0  
5                                                  ➕            0

In [81]:
comments_info.to_csv('comments_info.csv', index=False)